pkg/collector/autodiscovery/configresolver.go

```diff
@@ -158,14 +159,18 @@ func (cr *ConfigResolver) processNewService(svc listeners.Service) {
 	defer cr.m.Unlock()
 
 	// in any case, register the service
-	cr.services[svc.ID] = svc
-	cr.serviceToChecks[svc.ID] = make([]check.ID, 0)
+	cr.services[svc.GetID()] = svc
+	cr.serviceToChecks[svc.GetID()] = make([]check.ID, 0)
 
 	// get all the templates matching service identifiers
 	templates := []check.Config{}
-	for _, adID := range svc.ADIdentifiers {
+	ADIdentifiers, err := svc.GetADIdentifiers()
+	if err != nil {
+		log.Errorf("Failed to get AD identifiers from service - %s", err)
```

Comment:
We should tell users what's the impact of this: we won't be able to schedule checks for this container. Also we can probably return here?